pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -1096,10 +1127,18 @@ func (r *ReconcileVirtualMachineImport) updateProgress(instance *v2vv1.VirtualMa
 }
 
 func (r *ReconcileVirtualMachineImport) afterSuccess(vmName types.NamespacedName, p provider.Provider, instance *v2vv1.VirtualMachineImport) error {
+
+	err := utils.RemoveFinalizer(instance, utils.CancelledImportFinalizer, r.client)
```

Comment:
This code look the same as from line 1159. Please extract to a method.

Reply:
It is a call to a function and then short error handling. If we create a function for such duplication we'll end up with many functions, no?

Reply:
I noticed duplication maybe we can do it higher up the call chain to remove the need having it in both places.

Reply:
I'll do what you tell me, but I think it's OK to have duplicity. I do not know a way to avoid it completely. In this case, for example, these are operations we perform when import is done. If you look at the code that calls afterSuccess and afterFailure you'll see more duplicity (e.g. calling updateProgress(DONE)).

Reply:
I am open to other approaches. We could refactor the code and remove duplicity.

Reply:
I can add a new function: onImportDone({fail|success}).
Update progress.
Remove finalizer
Log
...